### DISCUSSION TITLE
Return true exit code when using tee to append the output to a log file in shell

### DISCUSSION THREAD

**Discussion by RedHoes** [ohmyzsh/ohmyzsh#12095] [Help] [answered]
We can use tee to capture real-time stdout or stderr and redirect it to a file
      ` ls -la 2>&1 | tee -a example.txt`

But I wonder how can this command return a true exit code, because by default, the tee command will return status code equal to 0, even the first command return an error:

```
$ ls -2 2>&1 | tee -a example.txt
ls: invalid option -- '2'
Try 'ls --help' for more information.
$ echo $?
0

```

* Note: Only use Linux **shell**  for this case

**Comment by mcornella** [comment, marked as answer]
See `$pipestatus`: https://unix.stackexchange.com/a/14276.

**Reply by RedHoes** [reply]
I want to use linux shell for this case instead of bash or zsh

**Reply by mcornella** [reply]
That StackExchange post has other options for POSIX shell.
